File: api/domain/d7d/sheet.go
package d7d

type Sheet struct {
	Name              string `json:"name"`
	Class             string `json:"class"`
	Race              string `json:"race"`
	Background        string `json:"background"`
	Alignment         string `json:"alignment"`
	PersonalityTraits string `json:"personality_traits"`
	Ideals            string `json:"ideals"`
	Bonds             string `json:"bonds"`
	Flaws             string `json:"flaws"`

	XP    uint32 `json:"xp"`
	Level uint8  `json:"level"`

	STR uint8 `json:"str"`
	DEX uint8 `json:"dex"`
	CON uint8 `json:"con"`
	INT uint8 `json:"int"`
	WIS uint8 `json:"wis"`
	CHA uint8 `json:"cha"`

	SaveSTR bool `json:"save_str"`
	SaveDEX bool `json:"save_dex"`
	SaveCON bool `json:"save_con"`
	SaveINT bool `json:"save_int"`
	SaveWIS bool `json:"save_wis"`
	SaveCHA bool `json:"save_cha"`

	Skills []Skill `json:"skills"`

	ArmorClass uint8  `json:"armor_class"`
	Speed      string `json:"speed"`

	Hp     uint8 `json:"hp"`
	HpMax  uint8 `json:"hp_max"`
	HpTemp uint8 `json:"hp_temp"`

	HitDice        string `json:"hit_dice"`
	HitDiceCurrent uint8  `json:"hit_dice_current"`
	HitDiceTotal   uint8  `json:"hit_dice_total"`

	DeathSavesSuccess uint8 `json:"death_saves_success"`
	DeathSavesFailure uint8 `json:"death_saves_failure"`

	Attacks []SheetAttack `json:"attacks"`

	Languages          string `json:"languages"`
	OtherProficiencies string `json:"other_proficiencies"`

	Age        uint8  `json:"age"`
	Height     string `json:"height"`
	Weight     string `json:"weight"`
	Eyes       string `json:"eyes"`
	Skin       string `json:"skin"`
	Hair       string `json:"hair"`
	Appearance string `json:"appearance"`
	Backstory  string `json:"backstory"`
}

type SheetAttack struct {
	Name             string `json:"name"`
	AttackDice       Dice   `json:"attack_dice"`
	AttackDiceAmount uint8  `json:"attack_dice_amount"`
	DamageType       string `json:"damage_type"`
}

File: web/app/domain/d7d/sheet/sheet.ts
import type { ESkill, TDice } from "../types";

export interface ISheetAttack {
    name: string;
    attack_dice: TDice;
    attack_dice_amount: number;
    damage_type: string;
}

export interface ISheet {
    id: number
    name: string
    class: string
    race: string
    background: string
    alignment: string
    personality_traits: string
    ideals: string
    bonds: string
    flaws: string
    xp: number
    level: number

    str: number
    dex: number
    con: number
    int: number
    wis: number
    cha: number

    save_str: boolean
    save_dex: boolean
    save_con: boolean
    save_int: boolean
    save_wis: boolean
    save_cha: boolean

    skills: ESkill[]

    armor_class: number
    speed: string

    hp: number
    hp_max: number
    hp_temp: number

    hit_dice: TDice
    hit_dice_current: number
    hit_dice_total: number

    death_saves_success: number
    death_saves_failure: number

    attacks: ISheetAttack[]

    languages: string
    other_proficiencies: string

    age: number
    height: string
    weight: string
    eyes: string
    skin: string
    hair: string
    appearance: string
    backstory: string
}
